nrf_security/src/psa/psa_eits_zephyr_settings.c

```diff
@@ -134,45 +176,270 @@ static psa_status_t is_write_once_set( psa_storage_uid_t uid,
+
+      case PATH_TYPE_IV:
+        ret = snprintk( buff, buff_size, "%s/%" PRIx64 "/iv", ROOT_KEY, uid );
+      break;
+
+      case PATH_TYPE_TAG:
+        ret = snprintk( buff, buff_size, "%s/%" PRIx64 "/tag", ROOT_KEY, uid );
+      break;
+    }
 
     __ASSERT( ret < buff_size, "Buffer size is too small." );
 }
 
-static void create_flags_path( psa_storage_uid_t uid, char *buff,
-                               size_t buff_size )
+static void fill_deriv_label(psa_storage_uid_t uid,
+                            uint8_t *deriv_label,
+                            size_t deriv_label_size)
 {
-    __ASSERT( buff != NULL, "Impropert value" );
+    size_t ret;
 
-    int ret = snprintk( buff, buff_size, "%s/%" PRIx64 "/flags", ROOT_KEY,
-                        uid );
-    (void)ret;
+    __ASSERT( deriv_label != NULL, "Improper value" );
+    __ASSERT( deriv_label_size >=  DERIV_LABEL_SIZE, "Improper value" );
 
-    __ASSERT( ret < buff_size, "Buffer size is too small." );
+    ret = snprintk(deriv_label, deriv_label_size, "%" PRIx64 "%s", uid, deriv_context);
+
+    __ASSERT( ret < expected_deriv_label_len, "Could not fill deriv_label." );
+}
+
+static void fill_add_data(psa_storage_uid_t uid,
+                         psa_storage_create_flags_t create_flags,
+                         uint32_t data_size,
+                         uint8_t *iv,
+                         size_t iv_len,
+                         uint8_t *add_data,
+                         size_t add_data_len)
+{
+    size_t buff_offset;
+
+    __ASSERT( add_data != NULL, "Improper value" );
+    __ASSERT( iv != NULL, "Improper value" );
+    __ASSERT( add_data_len >=  ENC_ADD_DATA_SIZE, "Improper value" );
+
+    buff_offset = 0;
+
+    memcpy(add_data, &uid, sizeof(uid));
+    buff_offset+=sizeof(uid);
+
+    memcpy(add_data + buff_offset, iv, iv_len);
+    buff_offset+=iv_len;
+
+    memcpy(add_data + buff_offset, &data_size, sizeof(data_size));
+    buff_offset+=sizeof(data_size);
+
+    memcpy(add_data + buff_offset, &create_flags, sizeof(create_flags));
+    buff_offset+=sizeof(create_flags);
+}
+
+static void zeroize_global_buffers(void)
+{
+    memset(encrypted_buffer, 0x0, CONFIG_PSA_EITS_READ_BUFF_SIZE);
+    memset(decrypted_buffer, 0x0, CONFIG_PSA_EITS_READ_BUFF_SIZE);
+    memset(deriv_label, 0x0, DERIV_LABEL_SIZE);
+    memset(tag, 0x0, ENC_TAG_SIZE);
+    memset(iv, 0x0, ENC_IV_SIZE);
+    memset(add_data, 0x0, ENC_ADD_DATA_SIZE);
+}
+
+static psa_status_t crypt_data(psa_storage_uid_t uid,
+                               psa_storage_create_flags_t create_flags,
+                               const uint8_t *input,
+                               uint32_t input_len,
+                               uint8_t *output,
+                               bool is_encrypt)
+{
+    nrf_cc3xx_platform_derived_key_ctx_t ctx;
+    int ret;
+
+    ret = nrf_cc3xx_platform_derived_key_init(&ctx);
+    if (ret != NRF_CC3XX_PLATFORM_SUCCESS){
+        return PSA_ERROR_STORAGE_FAILURE;
```

Comment:
Perhaps the errors should be more specific?

Reading: https://armmbed.github.io/mbed-crypto/html/api/library/status.html#error-codes

I see about PSA_ERROR_STORAGE_FAILURE

> There was a storage failure that might have led to data loss.

and

> It is recommended to only use this error code to report a permanent storage corruption. 

For this specific line there does not seem to be any best candidate though...
Maybe `PSA_ERROR_GENERIC_ERROR` or `PSA_ERROR_INVALID_ARGUMENT`